### PR TITLE
Npm and Java telemetry

### LINKED ISSUE #18800: Ensure JavaScript and Java test failures are properly categorized in build telemetry
Docs on build telemetry: https://github.com/dotnet/arcade/blob/30e2cca9e6f26a843bad0910879d89035645bd20/Documentation/Projects/DevOps/CI/Telemetry-Guidance.md

Right now, failures in Java tests and JavaScript tests show up as "Build" or "Unknown" failures. We need to add the necessary goop to categorize them as "Test" failures. This should be straightforward, looking at the docs above.

cc @dougbu @BrennanConroy

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspnetcore/issues/18800

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Previously a java test failure looked like:
`##[error]src\SignalR\clients\java\signalr\signalr.client.java.javaproj(55,5): error MSB3073: The command "./gradlew  -Dorg.gradle.daemon=false -PpackageVersion="5.0.0-ci" test" exited with code 1.`
Now:
`##[error]src\SignalR\clients\java\signalr\signalr.client.java.javaproj(58,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Test) The command "./gradlew  -Dorg.gradle.daemon=false -PpackageVersion="5.0.0-ci" test" exited with code 1.`

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/NegotiateResponseTest.java [hunk 1]
```diff
@@ -29,6 +29,7 @@ public void VerifyNegotiateResponse() {
         assertEquals("bVOiRPG8-6YiJ6d7ZcTOVQ", negotiateResponse.getConnectionId());
         assertEquals("connection-token-value", negotiateResponse.getConnectionToken());
         assertEquals(99, negotiateResponse.getVersion());
+        assertTrue(false);
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/NegotiateResponseTest.java]:
```suggestion
```

File: src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts [hunk 2]
```diff
@@ -76,6 +76,8 @@ describe("hubConnection", () => {
                     fail(e);
                     done();
                 });
+
+                fail(new Error());
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts]:
```suggestion
```

File: src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts [hunk 3]
```diff
@@ -76,6 +76,8 @@ describe("hubConnection", () => {
                     fail(e);
                     done();
                 });
+
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts]:
```suggestion
```

File: src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts [hunk 4]
```diff
@@ -38,6 +38,8 @@ describe("HubConnection", () => {
                 } finally {
                     await hubConnection.stop();
                 }
+
+                throw new Error();
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts]:
```suggestion
```

File: src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts [hunk 5]
```diff
@@ -38,6 +38,8 @@ describe("HubConnection", () => {
                 } finally {
                     await hubConnection.stop();
                 }
+
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts]:
```suggestion
```

**Review by analogrelay — APPROVED** [review]
Looks good!

**Comment by dougbu** [comment]
Thanks @BrennanConroy ❕

**Pull request merged**